### PR TITLE
Add default constructor for hp::DoFHandler.

### LINKED ISSUE #6259: Inconsistent interface between DoFHandler and hp::DoFHandler
`hp::DoFHandler` lacks a default constructor and member function `initialize`, is this intended?

### PR DESCRIPTION
**Pull request by adamqc**
This PR add a default constructor and a member function `initialize` to `hp::DoFHandler`, making the interface consistent with `DoFHandler`.

closes #6259.

### CODE REVIEW

File: source/hp/dof_handler.cc [hunk 1]
```diff
@@ -1269,6 +1276,40 @@ namespace hp
       active_fe_indices[i] = cell->active_fe_index();
   }
 
+  template <int dim, int spacedim>
+  void DoFHandler<dim,spacedim>::initialize(const Triangulation<dim,spacedim> &tria,
+                                            const hp::FECollection<dim,spacedim> &fe)
+  {
+    this->tria = &tria;
+
+    if (dynamic_cast<const parallel::shared::Triangulation< dim, spacedim>*> (&*this->tria) != nullptr)
```
Comment (bangerth) [inline source/hp/dof_handler.cc]: Most of this is copied from the other constructor -- could you break this common code out into a separate function to minimize the code duplication?

File: source/hp/dof_handler.cc [hunk 2]
```diff
@@ -959,6 +959,13 @@ namespace hp
...
 
 
+  template <int dim, int spacedim>
+  DoFHandler<dim,spacedim>::DoFHandler ()
+    :
+    tria(nullptr, typeid(*this).name()),
+    faces (nullptr)
+  {}
+
   template <int dim, int spacedim>
```
Comment (bangerth) [inline source/hp/dof_handler.cc]: Total nitpick, but we like to have three empty lines between each two function definitions.

**Comment by adamqc** [comment]
@bangerth This PR is ready to be reviewed again.

**Comment by bangerth** [comment]
Nice, very clean code!

OK to merge once the tester is happy.
/run-tests

**Comment by bangerth** [comment]
Oh, one more thing -- please add an entry to `doc/news/changes/minor` that describes the change.

**Comment by bangerth** [comment]
OK to merge once the tester is happy.

**Comment by davydden** [comment]
thanks @adamqc 👍

**Pull request merged**